syntax/fluent.ebnf

```diff
@@ -1,41 +1,53 @@
-body                 ::= (entry NL)* entry? EOF
+body                 ::= (_* NL)* (entry NL)* entry? EOF
 entry                ::= comment
                        | section
                        | message
 
-comment              ::= ('//' [^#xA#xD]* NL)+
-section              ::= '[[' __ word (__ word)* __ ']]'
+comment              ::= ('//' (char - NL)* )+
+section              ::= '[[' _ word (_ word)* _ ']]'
 
 char                 ::= [https://www.w3.org/TR/REC-xml/#NT-Char]
-__                   ::= [#x20#x9]*   /* space, tab */
-NL                   ::= [#xA#xD]+    /* line feed, carriage return */
+
+/* special chars: { } [ \ ] */
+special              ::= [#x5b#x5c#x5d#x7b#x7d]
+
+/* line feed, carriage return; space, tab */
+line-break           ::= [#xA#xD]+
+inline-space         ::= [#x20#x9]+
+
+/* if you break lines you need to indent afterwards */
+break-indent         ::= (line-break* inline-space)+
+
+_                    ::= inline-space
+NL                   ::= line-break
+__                   ::= break-indent
 
 identifier           ::= [a-zA-Z_?-] [a-zA-Z0-9_?-]*
 external             ::= '$' identifier
-word                 ::= [^#x20#x9#xA#xD#x5b#x5c#x5d]+   /* exclude: white space, [, \, ] */
+word                 ::= (((char - line-break) - inline-space) - special)+
 builtin              ::= [A-Z_?-]+
 number               ::= [0-9]+ ('.' [0-9]+)?
 
 variant-key          ::= number | variant-symbol
-variant-symbol       ::= word (__ word)*
-variant              ::= '[' variant-key ']' __ pattern NL
-default-variant      ::= '*' variant
-variant-list         ::= NL (__ variant)* __ default-variant (__ variant)*
-
-attribute            ::= '.' identifier __ '=' __ pattern NL
-attribute-list       ::= NL (__ attribute)+
-
-message              ::= identifier __ '=' __ (pattern attribute-list? | attribute-list)
-pattern              ::= unquoted-pattern
-                       | quoted-pattern
-unquoted-pattern     ::= (unquoted-text | placeable | block-text)+
-quoted-pattern       ::= '"' (quoted-text | placeable)* '"'
-unquoted-text        ::= ([^{] | '\{')+
-quoted-text          ::= ([^{"] | '\{' | '\"')+
-block-text           ::= NL __ '|' unquoted-pattern
-
-placeable            ::= '{' __ (expression | select-expression | variant-list) __ '}'
-expression           ::= quoted-pattern
+variant-symbol       ::= word (_ word)*
+variant              ::= NL __ '[' _ variant-key _ ']' __ pattern
+default-variant      ::= NL __ '*[' _ variant-key _ ']' __ pattern
+variant-list         ::= variant* default-variant variant*
+
+tag                  ::= '#' word
+tag-list             ::= NL (__ tag)+
+
+attribute            ::= NL __ '.' identifier value
+attribute-list       ::= attribute+
+
+message              ::= identifier ((value tag-list?) | (value? attribute-list))
+value                ::= __ '=' __ pattern
```

Comment:
so, this opens up
```
key
  = Value
```

I have to say I don't like it from the readability perspective :(